components/camel-google-bigquery/src/main/java/org/apache/camel/component/google/bigquery/GoogleBigQueryEndpoint.java

```diff
@@ -0,0 +1,105 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.google.bigquery;
+
+import java.util.concurrent.ExecutorService;
+
+import com.google.api.services.bigquery.Bigquery;
+import org.apache.camel.Consumer;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.impl.DefaultEndpoint;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+
+/**
+ * BigQuery Endpoint Definition
+ * Represents a table within a BigQuery dataset
+ * Contains configuration details for a single table and the utility methods (such as check, create) to ease operations
+ * URI Parameters:
+ * * Logger ID - To ensure that logging is unified under Route Logger, the logger ID can be passed on
+ *               via an endpoint URI parameter
+ * * Partitioned - to indicate that the table needs to be partitioned - every UTC day to be written into a
+ *                 timestamped separate table
+ *                 side effect: Australian operational day is always split between two UTC days, and, therefore, tables
+ *
+ * Another consideration is that exceptions are not handled within the class. They are expected to bubble up and be handled
+ * by Camel.
+ */
+@UriEndpoint(scheme = "bigquery", title = "BigQuery", syntax = "bigquery:projectId:datasetId[:tableName]", label = "messaging")
+public class GoogleBigQueryEndpoint extends DefaultEndpoint {
+    @UriParam
+    protected final GoogleBigQueryConfiguration configuration;
+
+    private final Bigquery bigquery;
+    private final String uri;
+
+    private ExecutorService executorService;
+
+    protected GoogleBigQueryEndpoint(String uri, Bigquery bigquery, GoogleBigQueryConfiguration configuration) {
+        this.bigquery = bigquery;
+        this.uri = uri;
+        this.configuration = configuration;
+    }
+
+    public Producer createProducer() throws Exception {
+        GoogleBigQueryProducer producer = new GoogleBigQueryProducer(this, configuration);
+        if (configuration.getConcurrentProducers() > 0) {
+            executorService = getCamelContext()
+                    .getExecutorServiceManager()
+                    .newFixedThreadPool(
+                            this,
+                            "camel-google-bigquery",
+                            configuration.getConcurrentProducers()
+                    );
+        } else {
+            executorService = getCamelContext()
+                    .getExecutorServiceManager()
+                    .newDefaultThreadPool(this, "camel-google-bigquery");
+        }
+        return producer;
+    }
+
+
+    public Consumer createConsumer(Processor processor) throws Exception {
+        throw new UnsupportedOperationException("Cannot consume from the BigQuery endpoint: " + getEndpointUri());
+    }
+
+    public boolean isSingleton() {
```

Comment:
Avoid non singleton at all cost. Instead you should store the local state on the producer/consumers, such as its executor service etc. The endpoint should generally just be a stateless class.